src/wasm/wasm-type.cpp

```diff
@@ -1340,10 +1388,29 @@ void Canonicalizer::makeAncestorsFixedPoint() {
   } while (changed);
 }
 
+void Canonicalizer::findRecursiveTypes() {
+  for (auto& kv : ancestors) {
+    TypeID curr = kv.first;
+    if (kv.second.count(kv.first) != 0) {
```

Comment:
```suggestion
    if (kv.second.count(curr) != 0) {
```
Also capturing `kv.second` to a local would give it a name which could be more readable I think.